File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	var result int

	ns := strconv.Itoa(n)
	ndigits := len(ns)

	for _, c := range ns {
		num, _ := strconv.Atoi(string(c))
		result += int(math.Pow(float64(num), float64(ndigits)))
	}

	return result == n
}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
  if (candidate == 0) {
    return true;
  }
  int number = candidate;
  int digits = 0;
  int result = 0;

  while (number > 0) {
    digits++;
    number /= 10;
  }

  number = candidate;
  while (number > 0) {
    int last_digit = number % 10;
    result += pow(last_digit, digits);

    number /= 10;
  }

  return result == candidate;
}
